### ISSUE TITLE
Transfer learning

### ISSUE THREAD

**Issue by thadpasce16** [ppwwyyxx/tensorpack#87]
Not an issue, a request for an example.

It would be helpful (for newbies) if you can provide an example of transfer learning. Say you start with Alexnet weights trained on imagenet (1000 classes) and then do transfer learning for CIFAR-10 data. This would change the last FC to output 10 classes.

I can think of how this can be done but not clear on how to not change the weights of the 5 CONV and 2 FC layers (the alexnet net), i,e, freeze their weights, but change weights of only the final FC layer's weight during fine tuning.

Thanks

**Comment by ppwwyyxx** [comment]
1. Loading a pre-trained model is just simply `--load model` handled in most examples. `--load` will try to load all parameters with the same name (and print warning about the ummatched weights in the model). [HED](https://github.com/ppwwyyxx/tensorpack/tree/master/examples/HED) is such example.

2. To freeze weights for the first several layers, you can just use [tf.stop_gradient](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/api_docs/python/functions_and_classes/shard3/tf.stop_gradient.md). To freeze only certain weights in between, you can use `ScaleGradient` to set a learning rate multiplier to those weights (e.g. in the [SpatialTransformer example](https://github.com/ppwwyyxx/tensorpack/blob/master/examples/SpatialTransformer/mnist-addition.py#L89).

**Comment by itmessager** [comment]
>     1. Loading a pre-trained model is just simply `--load model` handled in most examples. `--load` will try to load all parameters with the same name and print warning about the ummatched weights in the model, so to re-train some layers you should change their names before loading. [HED](https://github.com/ppwwyyxx/tensorpack/tree/master/examples/HED) is such an example which loads pre-trained vgg..
> 
>     2. [FAQ](http://tensorpack.readthedocs.io/en/latest/tutorial/faq.html#how-to-freeze-some-variables-in-training) about freezing variables.

can't find the FAQ page

**Comment by ppwwyyxx** [comment]
https://tensorpack.readthedocs.io/tutorial/faq.html

**Comment by itmessager** [comment]
> https://tensorpack.readthedocs.io/tutorial/faq.html

thanks
